src/runners/rfq_maker_balance_cache_runner.ts

```diff
@@ -0,0 +1,169 @@
+    tokens: string[];
+}
+
+if (require.main === module) {
+    (async () => {
+        logger.info('running RFQ balance cache runner');
+        const ethersProvider = new ethers.providers.JsonRpcProvider(
+            defaultHttpServiceWithRateLimiterConfig.ethereumRpcUrl,
+        );
+        const connection = await getDBConnectionAsync();
+        const balanceCheckerContractInterface = await _getBalanceCheckerContractInterfaceAsync(ethersProvider);
+
+        await runRfqBalanceCheckerAsync(ethersProvider, connection, balanceCheckerContractInterface);
+    })().catch(error => logger.error(error.stack));
+}
+
+async function runRfqBalanceCheckerAsync(
+    ethersProvider: ethers.providers.JsonRpcProvider,
+    connection: Connection,
+    balanceCheckerContractInterface: ethers.Contract,
+): Promise<void> {
+    const workerId = _.uniqueId('rfqw_');
+    let lastBlockSeen = -1;
+    while (true) {
+        const newBlock = await ethersProvider.getBlockNumber();
+        if (lastBlockSeen < newBlock) {
+            lastBlockSeen = newBlock;
+            LATEST_BLOCK_PROCESSED_GAUGE.labels(workerId).set(lastBlockSeen);
+            logUtils.log(
+                {
+                    block: lastBlockSeen,
+                    workerId,
+                },
+                'Found new block',
+            );
+
+            const makerTokens = await _getMakerTokensAsync(connection);
+            const balancesCallInput = _splitValues(makerTokens);
+
+            const updateTime = new Date();
+            const erc20Balances = await _getErc20BalancesAsync(balanceCheckerContractInterface, balancesCallInput);
+
+            await _updateErc20BalancesAsync(balancesCallInput, erc20Balances, connection, updateTime);
+
+            await delay(DELAY_WHEN_NEW_BLOCK_FOUND);
+        } else {
+            await delay(DELAY_WHEN_NEW_BLOCK_NOT_FOUND);
+        }
+    }
+}
+
+// NOTE: this only returns a partial entity class, just token address and maker address
+// Cache the query results to reduce reads from the DB
+let MAKER_TOKEN_CACHE: ResultCache<MakerBalanceChainCacheEntity[]>;
+const _getMakerTokensAsync = async (connection: Connection) => {
+    if (!MAKER_TOKEN_CACHE) {
+        logUtils.log('updating cache');
+        MAKER_TOKEN_CACHE = createResultCache<any[]>(
+            () =>
+                connection
+                    .getRepository(MakerBalanceChainCacheEntity)
+                    .createQueryBuilder('maker_balance_chain_cache')
+                    .select(['maker_balance_chain_cache.tokenAddress', 'maker_balance_chain_cache.makerAddress'])
+                    .getMany(),
+            TEN_MINUTES_MS,
+        );
+    }
+    return (await MAKER_TOKEN_CACHE.getResultAsync()).result;
+};
+
+function _splitValues(makerTokens: MakerBalanceChainCacheEntity[]): BalancesCallInput {
+    const functionInputs: BalancesCallInput = { addresses: [], tokens: [] };
+
+    return makerTokens.reduce(({ addresses, tokens }, makerToken) => {
```

Comment:
Just wanted to provide a suggested alternative implementation, however, your implementation is correct so don't feel you need to change anything unless you prefer :)

```
const addresses = [];
const tokens = [];
for (const makerToken of makerTokens) {
    addresses.push(makerToken.makerAddress);
    tokens.push(makerToken. tokenAddress);
}
return {tokens, addresses};
```

On another note, can we change the name `makerTokens` to be called something more like `balanceEntries`.

Reply:
discussed, leaving as is